transfer/transfer-06-http-data-flow/backend-service/src/main/java/org/eclipse/edc/handler/ReceiverHandler.java

```diff
@@ -0,0 +1,53 @@
+/*
+ *  Copyright (c) 2022 Bayerische Motoren Werke Aktiengesellschaft (BMW AG)
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Bayerische Motoren Werke Aktiengesellschaft (BMW AG) - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.handler;
+
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+import org.eclipse.edc.parser.DefaultHttpQueryParser;
+import org.eclipse.edc.parser.HttpQueryParserBase;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.URI;
+import java.util.HashMap;
+import java.util.Map;
+
+public class ReceiverHandler implements HttpHandler {
+
+    private final HttpQueryParserBase queryParser = new DefaultHttpQueryParser();
+
+    @Override
+    public void handle(HttpExchange exchange) throws IOException {
+        // parse request
+        Map<String, Object> parameters = new HashMap<>();
+        URI requestedUri = exchange.getRequestURI();
+        String query = requestedUri.getRawQuery();
+        queryParser.parseQuery(query, parameters);
+
+        System.out.println("Request Body: " + new String(exchange.getRequestBody().readAllBytes()));
+
+        // send response
+        StringBuilder response = new StringBuilder();
```

Comment:
Why build the response out of the input query, please just use a static string (also an empty one could be enough)